eng/pipelines/templates/jobs/archetype-go-release.yml

```diff
@@ -31,7 +31,6 @@ stages:
     jobs: 
       - deployment: TagRepository
         displayName: "Create release tag"
-        condition: and(succeeded(), eq(variables['NeedToRelease'], 'true'), ne(variables['Skip.TagRepository'], 'true'))
```

Comment:
We only want to remove the "NeedToRelease" part of this condition.

Reply:
`Skip.TagRepository` has already been checked with previous stage `CheckRelease` which is the dependency of this `Release` stage. So I removed it.

Reply:
Maybe I'm missing it but I don't see where `Skip.TagRepository` is checked. Can you point me to the place we check it?

Reply:
Sorry. I've mixed the two conditions (`Skip.Release`, `Skip.TagRepository`) up 😟. I've fixed it.